pkg/server/grpc_server.go

```diff
@@ -62,30 +62,38 @@ func newAPIserver(b *BgpServer, g *grpc.Server, hosts string) *server {
 func (s *server) serve() error {
 	var wg sync.WaitGroup
 	l := strings.Split(s.hosts, ",")
+	retries := 3
 	wg.Add(len(l))
 
-	serve := func(host string) {
+	serve := func(host string, retries int) {
 		defer wg.Done()
 		lis, err := net.Listen("tcp", host)
-		if err != nil {
+		for k := 1; k <= retries; k++ {
+			if err != nil {
+				log.WithFields(log.Fields{
+					"Topic": "grpc",
+					"Key":   host,
+					"Error": err,
+					"Retry": k,
+				}).Warn("listen failed")
+				lis.Close()
+				continue
+			}
+			err = s.grpcServer.Serve(lis)
```

Comment:
What's kinda situation could be handled by retrying this function?

Reply:
I fixed the conditions.
Sometimes listen failed it caused by timeout or the error is temporary , I think it can retry.